integration/testrunner/config.go

```diff
@@ -75,16 +67,48 @@ type TestCaseConfig struct {
 	TalkerClient talkerpb.TalkerClient
 }
 
+type privateKeyJSONData struct {
+	Encoded string
+}
+
+func parsePrivateKey(path string) (*ecdsa.PrivateKey, error) {
+	jsonFile, err := os.Open(path)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse private key: %w from file: %s", err, path)
+	}
+	b, err := io.ReadAll(jsonFile)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read data from key file: %w", err)
+	}
+	var data privateKeyJSONData
+	err = json.Unmarshal(b, &data)
+	if err != nil {
+		return nil, fmt.Errorf("failed to unmarshal to privateKeyJSONData: %w", err)
+	}
+	privateKeyPEM, _ := pem.Decode([]byte(strings.TrimSpace(data.Encoded)))
+	privateKey, err := x509.ParseECPrivateKey(privateKeyPEM.Bytes)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse EC private key: %w", err)
+	}
+	return privateKey, nil
+}
+
 func newTestConfig(ctx context.Context) (*Config, error) {
 	var c Config
 	if err := envconfig.ProcessWith(ctx, &c, envconfig.OsLookuper()); err != nil {
 		return nil, fmt.Errorf("failed to process environment: %w", err)
 	}
+	PrivateKey, err := parsePrivateKey(c.PrivateKeyFilePath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse private key: %w", err)
+	}
+	c.PrivateKey = *PrivateKey
```

Comment:
@capri-xiyue we should avoid dereferencing a pointer right IIRC, so c.PrivateKey = PrivateKey?